python/ray/_private/node.py

```diff
@@ -192,8 +193,16 @@ def __init__(
         # Initialize webui url
         if head:
             self._webui_url = None
+            self._webui_url_with_protocol = None
         else:
-            self._webui_url = ray._private.services.get_webui_url_from_internal_kv()
+            self._webui_url_with_protocol = (
+                ray._private.services.get_webui_url_from_internal_kv()
+            )
+            self._webui_url = (
+                self._remove_protocol_from_url(self._webui_url_with_protocol)
+                if self._webui_url_with_protocol
+                else None
```

Comment:
When is None returned? Can you comment that?

Reply:
Updated to return `self. _webui_url_with_protocol` if it is an empty string and doesn't need to be parsed